### PR TITLE
removing no data message from category descriptions & displaying descriptions correctly

### PR DESCRIPTION
**Pull request by emre2038**
## Description
this PR will replace https://github.com/OpenUpSA/wazimap-ng-ui/pull/427

We should not indicate "no data available" outside of an indicator

image

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-62

## How to test it locally
* go to SANEF profile
* expand rich data panel
* confirm that we don't show "No data available for this indicator for this geographic area" message outside indicators - especially in category/subcategory descriptions

### Updated
* `profile/subcategory.js` to remove the unnecessary error message
* `profile/category.js` & `profile/subcategory.js` to remove `.hidden` from description element when there is description to display
* `__tests__/profile/rich_data.test.js` to add a unit test

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/profile/subcategory.js [hunk 1]
```diff
@@ -92,20 +93,12 @@ export class Subcategory extends Component {
...
-                    
                     index++;
-                } else {
-                    $(wrapper).find(descriptionTextClass).text('No data available for this indicator for this geographic area');
-                    $(wrapper).find(descriptionClass).removeClass('hidden')
                 }
             }
-        } else {
-            $(wrapper).find(descriptionTextClass).text('No data available for this indicator for this geographic area');
-            $(wrapper).find(descriptionClass).removeClass('hidden')
```
Comment (michaelglenister) [inline src/js/profile/subcategory.js:108]: Why do we display an indicator if it doesn't have data. Maybe it is better to keep the message or not display the empty indicator?

Comment (michaelglenister) [reply src/js/profile/subcategory.js:108]: I double-checked, this change is correct. The mistake was that these are being displayed outside the indicators for categories and not indicators

**Comment by michaelglenister** [comment]
I'm happy with these changes

**Comment by michaelglenister** [comment]
Actually, there are some errors on the build preview now. I think best is to solve conflicts and check again

**Comment by jbothma** [comment]
@emre2038 just check out this error - perhaps it'll be fine once you merge staging back into this
Screenshot_2021-11-04_17-01-04

**Pull request merged**